lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1240,6 +1240,12 @@ void addIntelFPGADecorationsForStructMember(
     return;
 
   for (const auto &I : Decorations) {
+    // Such decoration already exists on a type, skip it
```

Comment:
What about the same check in `addIntelFPGADecorations` ?

Reply:
Good idea! Done.